federation/README.md

```diff
@@ -0,0 +1,67 @@
+# Cluster Federation
+
+Kubernetes Cluster Federation enables users to federate multiple
+Kubernetes clusters. Please see the [user guide](http://kubernetes.io/docs/admin/federation/)
+and the [admin guide](http://kubernetes.io/docs/user-guide/federation/federated-services/)
+for more details about setting up and using the Cluster Federation.
+
+# Building Kubernetes Cluster Federation
+
+Please see the [Kubernetes Development Guide](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md)
+for initial setup. Once you have the development environment setup
+as explained in that guide, you also need to install [`jq`](https://stedolan.github.io/jq/download/)
+<!-- TODO(madhusudancs): Re-evaluate using jq even in the development
+     environment. There is a concern that adding more tools as dependencies
+     might lead to proliferation of tools one need to install to develop
+     Kubernetes. jq is already a dependency for kubernetes-anywhere on
+     which this workflow depends, so we are giving an exception to jq
+     for now. -->
+
+Building cluster federation should be as simple as running:
+
+```shell
+make build do=gen
+```
+
+To deploy clusters and install federation components, edit the
+`config.default.json` file to describe your clusters and run
+
+```shell
+make build do=deploy
```

Comment:
Is do=gen a required step or can I directly run deploy if I want to deploy release images.
Also how can I specify the image that I want to use for federation (if I want to use a release image)?

(Am just asking for my curiosity. Its fine to say we dont support it right now, if thats the case).